crypto/fipsmodule/sha/internal.h

```diff
@@ -28,14 +28,29 @@ extern "C" {
 #define SHA3_PAD_CHAR 0x06
 #define SHA3_ROWS 5
 
+// SHAKE constants, from NIST FIPS202.
+// https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.202.pdf
+#define SHAKE_PAD_CHAR 0x1F
+#define SHAKE128_BLOCKSIZE (KECCAK1600_WIDTH - 128 * 2) / 8
+#define SHAKE256_BLOCKSIZE (KECCAK1600_WIDTH - 256 * 2) / 8
+
+// In the sponge construction, the width of the underlying 
+// function minus the rate (the block size in bits).
+// The |SHA3_MIN_CAPACITY_BYTES| corresponds to the 
+// lowest security level capacity for SHA3/SHAKE.
```

Comment:
```suggestion
// In the sponge construction, capacity is the width of the underlying function
// minus the rate. In case of SHA3 and SHAKE, the underlying function is Keccak
// and the rate is the block size of a specific instantiation of SHA3 or SHAKE.
// |SHA3_MIN_CAPACITY_BYTES| corresponds to the minimum capacity taking into
// account all the instantiations of SHA3 and SHAKE we implement: SHA3-256,
// SHAKE128, SHAKE256. Since the block size of SHAKE128 is the largest among
// those, we compute the minimum capacity by subtracting SHA128 block size from
// Keccak width.
```

Reply:
Thank you so much. I have fixed it. I added the source for the capacity definition as NIST FIPS202:
`// [NIST FIPS202] Capacity: In the sponge construction,`